TA-eclecticiq/bin/validator/validate.py

```diff
@@ -0,0 +1,464 @@
+            if content:
+                permissions = content.get(DATA).get(PERMISSIONS)
+            else:
+                response.status_code = STATUS_CODE_401
+
+        return permissions, response.status_code
+
+    def validate_user_permissions(self, url, api_token, proxy_settings):
+        """Get permission ids granted to user.
+
+        :return: missing_permissions
+        :rtype: set
+        """
+        permissions_of_user, status_code = self.get_user_granted_permissions(
+            url, api_token, proxy_settings
+        )
+        missing_permissions = []
+        if status_code not in [STATUS_CODE_200, STATUS_CODE_201]:
+            logger.error(USER_UNAUTHORIZED)
+        elif not permissions_of_user and status_code in [
+            STATUS_CODE_200,
+            STATUS_CODE_201,
+        ]:
+            logger.error(MISSING_READ_PERMISSIONS)
+            missing_permissions = READ_PERMSSIONS
+        else:
+            ids_of_user = ValidateAccount.get_permssion_ids(
+                permissions_of_user
+            )  # permission ids possesed by user
+            permissions_data = self.get_platform_permissions(
+                url, api_token, proxy_settings
+            )
+            if permissions_data:
+                ids_required_for_user = ValidateAccount.get_platform_permission_ids(
+                    permissions_data
+                )
+                user_authenticated, permission_ids = ValidateAccount.authenticate_user(
+                    ids_of_user, ids_required_for_user
+                )
+
+                if not user_authenticated:
+                    # check for missing permissions
+                    permissions_data = self.get_platform_permissions(
+                        url, api_token, proxy_settings
+                    )
+                    missing_permissions = ValidateAccount.get_permission_name_from_id(
+                        permissions_data, permission_ids
+                    )
+            else:
+                missing_permissions = READ_PERMSSIONS
+                status_code = STATUS_CODE_401
+        return missing_permissions, status_code
+
+    def send_request(self, url, api_token, proxy_settings):
+        """Send an API request to the URL provided with api token and proxy .sets the error message in UI and Log.
+
+        :param url: API URL to send request
+        :type url: str
+        :param api_token: API token for authentication
+        :type params: str
+        :param proxy_settings: proxy details to be included in the request
+        :type proxy: dict
+        :return: response if True else {}
+        :rtype: Response else dict
+        """
+        try:
+
+            headers = {"Authorization": f"Bearer {api_token}"}
+            response = requests.request(
+                GET,
+                url,
+                headers=headers,
+                verify=True,
+                proxies=proxy_settings,
+                timeout=40,
```

Comment:
read timeout value from constants

Reply:
This one time out is different than other requests. Hence we  cant read it from constants.